xbmc/music/infoscanner/MusicInfoScanner.cpp

```diff
@@ -112,7 +104,14 @@ void CMusicInfoScanner::Process()
         m_handle = dialog->GetHandle(g_localizeStrings.Get(314));
     }
 
-    m_bClean = g_advancedSettings.m_bMusicLibraryCleanOnUpdate;
+    // check if we only need to perform a cleaning
+    if (m_bClean && m_pathsToScan.empty())
+    {
+      CleanDatabase(false);
+      m_bRunning = false;
+
+      return;
```

Comment:
Why do that? It is within an If statement where we have `m_pathsToScan.empty()`, so there are no items to scan, nothing to do but clean. Hence we clean and return. Or are you seeing something else?

BTW thnaks for looking.